### PR TITLE
Final outfit datums things

### PR DESCRIPTION
**Pull request by ShiftyRail**
Last thing I'm going to do on this branch, it basically replaces the old deprecated `/obj/abstract/loadout` thing and finishes porting outfit datums to strike teams.

One direct advantage is that now you can use `equip_loadout` to convert someone into an engineer, doctor, captain, etc... in one click with the correct ID, PDA, access, and everything. This previously wasn't possible.

This also re-arranges a bit the old `loadout` datums so that they are less clunky to use and won't result in items dropped at the player's feet.

Once this is merged, all there is left to do is to solve merge conflicts and the update will be ready.

I decided to delay a bit using outfit datums for suit-dispensers who equip people. But it's still on the plans :tm:

### CODE REVIEW

File: code/datums/outfit/special_outfits/special.dm [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+	outfit_name = "Soviet rigsuit"
+	items_to_spawn = list(
+		"Default" = list(
+			slot_wear_suit_str = /obj/item/clothing/suit/space/rig/soviet,
+			slot_head_str = /obj/item/clothing/head/helmet/space/rig/soviet,
+		),
+	)
+
+/datum/outfit/special/dread_gear
+	outfit_name = "Judge Dread"
```
Comment (boy2mantwicethefam) [inline code/datums/outfit/special_outfits/special.dm]: Well actually it's Judge Dredd

File: code/datums/outfit/outfit.dm [hunk 2]
```diff
@@ -82,7 +84,7 @@
 	return
 
 /datum/outfit/proc/pre_equip_disabilities(var/mob/living/carbon/human/H, var/list/items_to_equip)
-	if (H.client.IsByondMember())
+	if (H.client && H.client.IsByondMember())
```
Comment (jknpj) [inline code/datums/outfit/outfit.dm]:
```suggestion
	if (H.client?.IsByondMember())
```

File: code/game/objects/items/weapons/cards_ids.dm [hunk 3]
```diff
@@ -834,4 +850,4 @@
 	access = get_all_accesses()
 	access += get_all_centcom_access()
 
-/obj/item/weapon/card/id/death_commando_
\ No newline at end of file
+/obj/item/weapon/card/id/death_commando_
```
Comment (DamianX) [inline code/game/objects/items/weapons/cards_ids.dm]: I don't think this is supposed to have a _ at the end
it also seems to not have any variables defined?

Comment (ShiftyRail) [reply code/game/objects/items/weapons/cards_ids.dm]: it's a mystery
will purge

File: code/datums/outfit/outfit.dm [hunk 4]
```diff
@@ -109,14 +112,17 @@
...
-		if (snowflake_items && (slot in snowflake_items[H.mind.role_alt_title])) // ex: special_snowflakes["Vox"]["Emergency responder"].
+		if (snowflake_items && (slot in snowflake_items[H?.mind.role_alt_title])) // ex: special_snowflakes["Vox"]["Emergency responder"].
 			special_equip(H.mind.role_alt_title, slot, 	H)
 			continue
 
 		var/obj_type = L[slot]
 		if (islist(obj_type)) // Special objects for alt-titles.
 			var/list/L2 = obj_type
-			obj_type = L2[H.mind.role_alt_title]
+			if (H.mind && H.mind.role_alt_title in L2)
```
Comment (DamianX) [inline code/datums/outfit/outfit.dm:122]: BYOND is tarded and this needs to be written as 
```suggestion
			if (H.mind && (H.mind.role_alt_title in L2))
```

**Comment by DeityLink** [comment]
Hey wasn't that planned like years ago?

**Pull request merged**